### PR TITLE
pythonPackages.pykde4: init at 4.14.3

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Was removed due https://github.com/NixOS/nixpkgs/issues/24548 but it builds now. And it should stay as long as possible, because pykde5 bindings were never finished!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Do these bindings work with Plasma and/or the newer KDE applications? Are you using these bindings?

File: pkgs/desktops/kde-4.14/kdebindings/pykde4.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ kde, kdelibs, pythonPackages, kdepimlibs, shared_desktop_ontologies,
```
Comment (FRidh) [inline pkgs/desktops/kde-4.14/kdebindings/pykde4.nix]: Python bindings are just Python libraries and so should therefore be called from `python-packages.nix`, with the expression also living in `python-modules`.

**Comment by gnidorah** [comment]
@FRidh I'm using only KStatusNotifierItem and KMenu from them. Just tested, yes, these parts work under Plasma 5.

File: pkgs/desktops/kde-4.14/kdebindings/pykde4.nix (hunk 1 above)
Comment (gnidorah) [reply pkgs/desktops/kde-4.14/kdebindings/pykde4.nix]: @FRidh Done, thank you!

**Comment by gnidorah** [comment]
Going to try fix DLFCN error for >= python3.6

**Comment by gnidorah** [comment]
Done

File: pkgs/development/python-modules/pykde4/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{ kdelibs4, pythonPackages, kdepimlibs, shared_desktop_ontologies,
```
Comment (FRidh) [inline pkgs/development/python-modules/pykde4/default.nix]: can you list the actual dependencies instead of the package set

Comment (gnidorah) [reply pkgs/development/python-modules/pykde4/default.nix]: @FRidh This way?

**Comment by ttuegel** [comment]
> And it should stay as long as possible, because pykde5 bindings were never finished!

Which packages depend on `pykde4`?

**Comment by gnidorah** [comment]
@ttuegel Don't know, but any Python script which uses KStatusNotifierItem depends on this https://github.com/search?l=Python&q=KStatusNotifierItem%2Bpykde4&type=Code&utf8=%E2%9C%93

**Comment by ttuegel** [comment]
> Don't know, but a lot of Python scripts use pykde4

Are any of them in Nixpkgs? I'm going to remove kdelibs4 in about 30 days when upstream ends all support. What is the case for keeping it until then?

**Comment by gnidorah** [comment]
> Are any of them in Nixpkgs? I'm going to remove kdelibs4 in about 30 days when upstream ends all support. What is the case for keeping it until then?

I believe not, sorry. Just using pykde4 myself for scripts. I will try to make this package standalone (if possible) and disable it on Hydra, otherwise will just keep this package and kdelibs4 locally.

**Comment by gnidorah** [comment]
@ttuegel Ok, done

File: pkgs/development/python-modules/pykde4/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  nativeBuildInputs = [ cmake automoc4 ];
+
+  propagatedBuildInputs = [ pyqt4 ];
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    platforms = platforms.linux;
+    description = "Python bindings for KDE";
+    license = "GPL";
```
Comment (ttuegel) [inline pkgs/development/python-modules/pykde4/default.nix]: Could we use a license or licenses from `lib.licenses`? `"GPL"` is ambiguous.

Comment (gnidorah) [reply pkgs/development/python-modules/pykde4/default.nix]: @ttuegel Done, thanks!

**Comment by gnidorah** [comment]
@GrahamcOfBorg build python3Packages.pykde4

**Pull request closed without merging**